### PR TITLE
Feature compose -schema remove and filter 

### PR DESCRIPTION
**Pull request by MerinRaj-212052**
completed remove functionality in schema .
Filtered unique values in schema import

### CODE REVIEW

File: src/components/ComposePage/ModalBox/modalBox.tsx [hunk 1]
```diff
@@ -26,48 +32,55 @@ interface MyModalProps {
...
+    const uniqueValues = [...projectSchemaInfo, ...selectedTableArray].filter(
+      (item, index, self) => index === self.findIndex((t) => t.uid === item.uid)
+    );
 
-  const combinedArray: any = [...projectSchemaInfo, ...selectedTableArray];
+    setCombinedArray(uniqueValues);
+  }, [projectSchemaInfo, selectedTableArray]);
 
-  const tableUidArray = combinedArray.map((table:any) => parseInt(table.uid)); //taking uid's of selected tables
+  const tableUidArray = combinedArray.map((table: any) => parseInt(table.uid)); //taking uid's of selected tables
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/ModalBox/modalBox.tsx]: Always include conditional check with ?

File: src/components/ComposePage/ModalBox/modalBox.tsx [hunk 2]
```diff
@@ -26,48 +32,55 @@ interface MyModalProps {
...
+  const handleImport = () => {
     const requestBody = {
-      projectUid: 3,
+      projectUid: 1,
       sourceTableUids: tableUidArray,
     };
-    dispatch(postProjectSchemaInfoRequest(requestBody.projectUid, requestBody.sourceTableUids));
+    dispatch(
+      postProjectSchemaInfoRequest(
+        requestBody.projectUid,
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/ModalBox/modalBox.tsx]: Conditional check needed

File: src/components/ComposePage/ModalBox/modalBox.tsx [hunk 3]
```diff
@@ -26,48 +32,55 @@ interface MyModalProps {
...
-    try {
-      const response = await axios.delete(
-        "http://localhost:8080/sourcecourse/project-tables",
-        { data: requestBody }
-      );
-    } catch (error) {
-      console.error(error);
-    }
+    setCombinedArray((prevArray) =>
+      prevArray.filter((item) => item.uid !== uid)
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/ModalBox/modalBox.tsx]: Conditional check needs to betaken care in all the changed files

**Review by sourcecoursepoc — CHANGES_REQUESTED** [review]

File: src/components/ComposePage/ModalBox/modalBox.tsx (hunk 2 above)
Comment (MerinRaj-212052) [reply src/components/ComposePage/ModalBox/modalBox.tsx]: done

File: src/components/ComposePage/ModalBox/modalBox.tsx (hunk 1 above)
Comment (MerinRaj-212052) [reply src/components/ComposePage/ModalBox/modalBox.tsx]: done

File: src/components/ComposePage/ModalBox/modalBox.tsx (hunk 3 above)
Comment (MerinRaj-212052) [reply src/components/ComposePage/ModalBox/modalBox.tsx]: done

**Review by sourcecoursepoc — APPROVED** [review]

**Review by sourcecoursepoc — APPROVED** [review]

**Pull request merged**